### DISCUSSION TITLE
Vue3 project. Is there a way I can tell whether or not the wallet is connected

### DISCUSSION THREAD

**Discussion by s3mi0tics** [WalletConnect/walletconnect-monorepo#4528] [Developer Support] [answered]
I would like to change the UI of my application if there is a successful wallet connection. Is there a way I can add a listener event that will tell me whether or not the wallet has created a successful connection? I notice the button changes.

**Comment by ghost** [comment, marked as answer]
@s3mi0tics,  Web3Modal provides the hook that can check the wallet is connected or not.

`
import { useWeb3ModalAccount } from '@web3modal/ethers/vue' 

const { address, chainId, isConnected } = useWeb3ModalAccount()
`

Based on the isConnected status, you can update the button or not.
